pandas/core/dtypes/cast.py

```diff
@@ -1454,7 +1454,7 @@ def _ensure_nanosecond_dtype(dtype: DtypeObj) -> DtypeObj:
 
 # TODO: other value-dependent functions to standardize here include
 #  dtypes.concat.cast_to_common_type and Index._find_common_type_compat
-def find_result_type(left: ArrayLike, right: Any) -> DtypeObj:
+def find_result_type(left: ArrayLike, right: Any, strict_na: bool = False) -> DtypeObj:
```

Comment:
im wary of introducing this here.  is there a hurry on this or can i take some time to catch up on the recent PRs that got us here?

Reply:
The addition here is where the regression occurred fixing an Interval issue. Are you thinking of pushing the fix further down into `is_valid_na_for_dtype` or `ensure_dtype_can_hold_na` or adding more special casing/code duplication in Block.replace? 

As I see it, whichever way it's done there is always a risk of changing behavior whereas the fix here keeps the fix for the interval issue and also keeps the existing behavior for replace.

This is on master and I'm quite happy with follow-ups to this that refactor as it's easier to revert a refactor is there is an issue down the line whereas if we need to revert a regression fix we end up re-introducing the regression.

Reply:
> is there a hurry on this or can i take some time to catch up on the recent PRs that got us here?

no hurry (unless there are more refactors to this area of the code which make it more difficult to fix later)